server/src/test/java/com/google/coffeehouse/servlets/CreatePersonServletTest.java

```diff
@@ -0,0 +1,119 @@
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+/**
+ * Unit tests for {@link CreatePersonServlet}.
+ */
+public class CreatePersonServletTest {
+  private static final String NICKNAME = "Tim";
+  private static final String EMAIL = "test@fake.fake";
+  private static final String PRONOUNS = "he/him";
+  private static final String IDENTIFICATION_STRING = "predetermined-identification-string";
+
+  private CreatePersonServlet CreatePersonServlet;
+	private final LocalServiceTestHelper helper = new LocalServiceTestHelper();
+  private StringWriter stringWriter = new StringWriter();
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  
+  @Before
+  public void beforeTest() throws IOException {
+    helper.setUp();
+
+    IdentifierGenerator idGen = Mockito.mock(IdentifierGenerator.class);
+    Mockito.when(idGen.generateId()).thenReturn(IDENTIFICATION_STRING);
+    CreatePersonServlet = new CreatePersonServlet(idGen);
+
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class);
+    Mockito.when(response.getWriter()).thenReturn(new PrintWriter(stringWriter));
+  }
+
+  @After
+  public void afterTest() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void allSpecified() throws IOException {
+    Mockito.when(request.getParameter("nickname")).thenReturn(NICKNAME);
+    Mockito.when(request.getParameter("email")).thenReturn(EMAIL);
+    Mockito.when(request.getParameter("pronouns")).thenReturn(PRONOUNS);
+
+    CreatePersonServlet.doPost(request, response);
+    String result = stringWriter.toString();
+
+    // use multiple contains because order of json should not be relied upon per GSON documentation
+    Assert.assertTrue(
+        result.contains("\"email\":\"" + EMAIL + "\"")
+        && result.contains("\"nickname\":\"" + NICKNAME + "\"")
+        && result.contains("\"pronouns\":\"" + PRONOUNS + "\"")
+        && result.contains("\"userId\":\"" + IDENTIFICATION_STRING + "\"")
+    );
+  }
+
+  @Test
+  public void noPronounsSpecified() throws IOException {
+    Mockito.when(request.getParameter("nickname")).thenReturn(NICKNAME);
+    Mockito.when(request.getParameter("email")).thenReturn(EMAIL);
+
+    CreatePersonServlet.doPost(request, response);
+    String result = stringWriter.toString();
+
+    // use multiple contains because order of json should not be relied upon per GSON documentation
+    Assert.assertTrue(
+        result.contains("\"email\":\"" + EMAIL + "\"")
+        && result.contains("\"nickname\":\"" + NICKNAME + "\"")
+        && result.contains("\"userId\":\"" + IDENTIFICATION_STRING + "\"")
+    );
+
+    // pronouns were not specified, so they should not be in JSON representation
```

Comment:
Nit: Please update this file to use consistent capitalization of json/JSON.

Reply:
Good catch, this has been fixed.